### ISSUE TITLE
SEP V013 : Multi-source / multi-sink arrows

### ISSUE THREAD

**Issue by jakebeal** [SynBioDex/SBOL-visual#47]
We have not yet determined what it means when an arrow has multiple sources or multiple sinks.
The fundamental ambiguity in drawings at present is that this is currently used to mean two things:
1. One species that has multiple sources or multiple points of effect.
2. A chemical reaction that combines multiple species into one or splits one species into many

**Comment by chofski** [comment]
Draft of the SEP has been uploaded and comments very welcome. See: https://github.com/SynBioDex/SBOL-visual/blob/develop/SEPs/SEP_V013.md

**Comment by shyambhakta** [comment]
I think it should be made explicit whether species composition and decomposition glyphs (filled circle at junction) can be used with multiple sources or targets of an effect to indicate that composition or decomposition is necessary for the effect.
Would the top be allowed for indicating repression of the target P1 only after the products of R1 and R2 interact, as they would if they form a complex?
And would the bottom be allowed for indicating that protein R1 must be cleaved in order for the cleavage products to repress the target promoters?
![image](https://user-images.githubusercontent.com/5035245/45394970-53126a00-b5f9-11e8-91d0-c551abfa4b99.png)

Also, would it be permissible for the composition/decomposition glyphs be used simultaneously, as in to show interaction and exchange of something? Below, a protein complex is delivering a component to an RNA.
![image](https://user-images.githubusercontent.com/5035245/45395092-0ed39980-b5fa-11e8-81d0-7435d7d38454.png)

**Comment by cjmyers** [comment]
The first example looks fine.  I'm not as sure though about the second two.  I'm not sure these are biologically common examples.

**Comment by shyambhakta** [comment]
The second example I don't know of occurring.
But the third, the exchange, is… not uncommon, I think. But it would require some prior energy investment like phosphorylation, released to alter conformations and affinity, favoring the exchange of the binding partner. The examples I'm thinking of don't necessarily have a trimolecular intermediate…
Ah what about one RNA outcompeting another in binding to a target, a form of sRNA-mediated regulation. That's gotta be genuinely trimolecular.

I guess with these I was just wondering whether it's worthwhile to mention whether combinations of the examples are permissible or not.

**Comment by cjmyers** [comment]
One issue that I think needs considering is the third case uses a square process box for multiple reactants and multiple products in SBGN.  The second case (disassociation) uses a double circle.  Only association in SBGN uses the filled circle.  See sbgn.org.

**Comment by shyambhakta** [comment]
SBGN seems to allow the square process glyph to be generically used for [association and dissociation](https://sbgn.github.io/sbgn/symbols#pd-complex) as well as for transport and [energetic/conformational changes](https://sbgn.github.io/sbgn/symbols#pd-stimulation) (de/activation), at least in their symbol highlights page.
The above _broad_ usages of the square process glyph aren't specified in the [manual §2.3.1, 2.3.4–2.3.5](https://github.com/sbgn/process-descriptions/blob/master/UserManual/sbgn_PD-level1-user-public.pdf); it doesn't read that association and dissociation are more specific "processes" that can be specified with the square glyph, but if we are to assume this from the symbol highlights page, then:
• The filled circle should be the _specific_ process arc glyph for association.
• The empty double circle should be the _specific_ process arc glyph for dissociation.
• The square glyph should be the _generic_ process arc glyph, which I think we're allowing to omit, since I don't see it anywhere in SBOL. SBOL SEPs show targets of arrows pointing at arc lines directly instead of pointing at a process glyph upon the arc like required(?) in SBGN.

If the generic process glyph is only recommended but not required wherever there are arcs, then would the specific association/dissociation glyphs only be recommended? Or should the specific processes of association/dissociation require the glyphs, but not other kinds of processes?

**Comment by jakebeal** [comment]
I would agree with @shyambhakta that we should import the SBGN-PD glyphs wholesale, which would make square the generic process glyph rather than circle.

I also think the glyphs should be required rather than recommended, since otherwise we aren't actually disambiguating when there is no process glyph.

**Comment by jakebeal** [comment]
I've updated per this discussion, discussion on the list, and at COMBINE.  The diff is pending in pull request: #54

**Comment by JS3xton** [comment]
Would the following SBGN Process Node vs. SBOL Process Glyph representations be synonymous? Is one preferred over the other?
<br>

![split_arrows](https://user-images.githubusercontent.com/2179825/46822078-15b41000-cd50-11e8-9fbb-f96eb820a519.png)

**Comment by jakebeal** [comment]
Those glyphs are _not_ synonymous.

**Comment by cjmyers** [comment]
The process node is for multi-input/multi-output processes.  Consider a reaction that takes A+B to form C+D.  Also, the Association node in your picture above should be about the same size as the disassociation, ideally.

**Comment by shyambhakta** [comment]
The right column SBOL process glyphs show _either_ the circle _or_ triangle individually becoming a square, or the square becoming _either_ a circle _or_ a triangle. Both in separate processes, versus the SBGN process nodes showing a single process of combination/dissociation, which requires the 2:1 or 1:2 ratio of reactants:products.

**Comment by palchicz** [comment]
Before attempting to move this to a vote again, I'm seeing two lingering issues with the clarity of this SEP.

First, raised by @JS3xton, is the use of the word "process glyph". 

> The twice-used phrase "the junction MUST be annotated with a process glyph, as per SBGN" is unclear to me.
> 
> SBOL Visual 2.0 has a Process glyph, but that does not appear to be what the phrase is referring to. Rather, it appears to be referring to the association and dissociation SBGN Process Nodes? (I'm less familiar with SBGN). Clarifying that could be helpful in understanding the question.

If I understand your concern correctly, the issue is that "process glyph" has a pretty general meaning in both SBML and SBOL, but it is being used here to refer to specific types of processes: namely "association" and "dissociation" . If the language was changed from "the junction MUST be annotated with a process glyph, as per SBGN" to "the junction MUST be annotated with the process description glyph for [association|dissociation], as per SBGN", would that clear things up?

Second, raised by @cjmyers is omission of the simultaneous multi-source/multi-sink example from the SEP, which requires a square box. 

>missing the multi-source/multi-sink example which uses the square box process node from SBGN

@jakebeal or @chofski, could you respond to either/both of these concerns.

**Comment by JS3xton** [comment]
@palchicz
I see the language in question is coming right out of the SEP. My thoughts after rereading the SEP:

I would like to keep the SBOL Visual spec as simple as possible and as independent as possible. As such, I think we should just define two new SBOL Visual Interaction Glyphs: `Association` and `Dissociation`. `Association` is associated with SBO:0000177 and `Dissociation` is associated with SBO:0000180.
Reference to SBGN should be limited to the "Associated SBO term(s)" section of the specification (as is the case with the `Inhibition`, `Control`, and `Stimulation` SBOL Visual Glyphs).

SEP V013 should be updated to explicitly introduce the `Association` and `Dissociation` SBOL Glyphs (the examples already look great me). It might also help to explicitly categorize the "Multiple sources of production" section and the "Multiple points of effect" section as specific to the `Process` SBOL Visual glyph. In fact, I would be curious to know how those sections will manifest as changes to the spec (Notes on the `Process` glyph?).

Once the SEP is updated, I would expect the text of the voting form to be updated to reflect it. The first two questions might more explicitly reference the `Process` SBOL glyph, for example, and the last two questions might look like "Do you support adoption of the Association glyph? The Association glyph indicates that multiple source elements combine to produce a new species or act together for regulation", etc.

**Comment by JS3xton** [comment]
Regarding the multi-source/multi-sink square process glyph mentioned by @cjmyers, it's still not clear to me what is being proposed. Whatever it is, I think it should be distilled down to the proposal of a new glyph (e.g. a *new* SBOL Visual glyph for `multi-source/multi-sink Process`) or a concrete modification of an existing glyph (e.g. changing how the existing `Process` SBOL Visual glyph is illustrated).

And indeed, the SEP currently makes no mention of this either way, so it should be updated if modifications to the representation of multi-source/multi-sink processes are to be considered.

**Comment by jakebeal** [comment]
Based on these conversations, I have prepared an updated version of SEP V013, along with draft changes to the spec.  The pull request containing the full diffs is here: https://github.com/SynBioDex/SBOL-visual/pull/55

The updated SEP V013 in particular may be found here: https://github.com/SynBioDex/SBOL-visual/blob/398c3d7/SEPs/SEP_V013.md

**Comment by shyambhakta** [comment]
@jakebeal The process glyph example reads the same as the association glyph example: "Association of gRNA and Cas9 to form an active CRISPR complex." Though correct, perhaps it can mirror the circuit example given and say something like "Phosphorylation of an inactive transcription factor by a kinase to form an active transcriptional activator"
Also, typo "gylph" three times.

In the dissociation glyph spec image, should the inner circle be grayed in as it is? I would think that this indicates (based on the association glyph spec) that the inner circle must be filled in, unlike the SBGN glyph.

**Comment by jakebeal** [comment]
@shyambhakta Fixed the "gylph" -> "glyph"; thanks for the catch.   I've also updated the process glyph example text.

For the dissociation glyph spec: the grey circle means it _may_ be filled, and that if one chooses to do so it should be the inner circle and not both.  SBGN does not address the question of color fills one way or another.  Updated SEP at: https://github.com/SynBioDex/SBOL-visual/blob/6fbc027/SEPs/SEP_V013.md

**Comment by cjmyers** [comment]
Not sure why an association glyph cannot be used in your process glyph example.  Would be nice to have an example with actual multiple products.

**Comment by jakebeal** [comment]
@cjmyers It's not an association, it's a phosphorylation interaction that modifies the original without consuming the interacting species.

**Comment by cjmyers** [comment]
Ok, that makes sense then.

**Comment by palchicz** [comment]
Is it a concern that SBOL Visual already defines a "Process" glyph? I think this point has been brought up before, but I'm unclear if there was any resolution.

**Comment by jakebeal** [comment]
@palchicz The overlap is not a problem and is, in fact, intentional. Both of these are, in fact, grounded in the exact same SBO term.  The "node" version and "edge" version combine to allow multi-input and multi-output processes.

**Comment by jakebeal** [comment]
Accepted and integrated, and thus closed per SBOL procedure in updated SEP 001.
